### ISSUE TITLE
Question about review data and testing

### ISSUE THREAD

**Issue by camer314** [adeshpande3/LSTM-Sentiment-Analysis#27]
With my trained model I am able to get the correct sentiment out of short sentences such as "The movie sucked" but i assumed that if i passed in the entire contents of one of the review files, especially one it trained on, then it should also get the right sentiment, however it is woefully inaccurate.

What am i missing here? Can i not pass an arbitrary length (up to 250) block of text such as the content of any one of the files into getSentenceMatrix()?

For example, if i copy the code from the training script that runs the testing loop but instead use batches from the training set, as below:

```
ids = np.load(os.path.join( local_path, 'idsMatrix.npy'))

iterations = 10
for i in range(iterations):
    nextBatch, nextBatchLabels = getTrainBatch();
    print("Accuracy for this batch:", (sess.run(accuracy, {input_data: nextBatch, labels: nextBatchLabels})) * 100)

```
The results are less than stellar, especially as its using the training data:

```
Accuracy for this batch: 58.3333313465
Accuracy for this batch: 45.8333343267
Accuracy for this batch: 45.8333343267
Accuracy for this batch: 50.0
Accuracy for this batch: 50.0
Accuracy for this batch: 45.8333343267
Accuracy for this batch: 45.8333343267
Accuracy for this batch: 62.5
Accuracy for this batch: 50.0
Accuracy for this batch: 50.0
```

**Comment by adeshpande3** [comment]
My initial thought is that I believe I trained the network for too long and didn't add enough regularization in the model, and thus the saved models that I included might have overfit to the training dataset. I'd recommend splitting the dataset into test/train sets so that you can take a look at how the network performs on test strings, as it is training. You could also decrease the number of hidden units/layers. Also, I was discussing this topic with another friend and we realized that using an LSTM for sentiment analysis might actually be the wrong type of model. An LSTM network is very powerful in terms of what it can learn, and it turns out that a simpler model would probably be a better fit for this particular binary classification task.

(Sorry I didn't mention all these caveats in the tutorial! Been learning a lot more about ML and best practices in the past year or so)

**Comment by camer314** [comment]
Ignore everything i said, looking at your accuracy in Tensorboard i got curious as to why mine wasnt performing the same, went back to the original training script before i hacked away at it and then ended up with the same chart. As a result my testing script, taking random files as above now achieves accuracy in the region of 90%

The only mystery now is what did i do to it to screw it so badly...

**Comment by adeshpande3** [comment]
Ah interesting, so I guess maybe overfitting wasn't the issue. Does it perform correctly on short inputs? Also, do let us know if you figure out what might have went wrong. Always constructive to see the different reasons models lose accuracy.

**Comment by camer314** [comment]
Short inputs seem to be ok, in my limited testing. I am not very creative when it comes to thinking of things to pass in :)

Is there any significance in the weightings on the output? Positive is when [0] is greater than [1] but can anything further be derived, such as confidence?

**Comment by adeshpande3** [comment]
Not really, you could put a softmax layer after the output in order to normalize the values into probabilities, but can't really obtain a confidence measure.

**Comment by camer314** [comment]
softmax did the trick, happy with that
